torch/fx/experimental/proxy_tensor.py

```diff
@@ -821,28 +852,24 @@ def get_torch_dispatch_modes():
 
 
 def get_innermost_proxy_mode():
-    for m in reversed(torch.utils._python_dispatch._get_current_dispatch_mode_stack()):
-        if isinstance(m, ProxyTorchDispatchMode):
-            return m
-    return None
+    return torch._C._get_dispatch_mode(torch._C.TorchDispatchModeKey.PROXY)
 
 
 @contextlib.contextmanager
 def disable_proxy_modes_tracing(enable_current=False):
```

Comment:
I think you can do some prep refactor.  The only use of enable_current=True is in `torch/_dynamo/compiled_autograd.py` and I think it is unnecessary.

Once you get rid of this arg, this function is literally "disable the proxy mode slot" which is very simple.